File: c++/troco_verificado.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {

    int qtd;
    double preco, dinheiro, troco, valorProduto;

    cout << "Preço unitário do produto: ";
    cin >> preco;
    cout << "Quantidade comprada: ";
    cin >> qtd;
    cout << "Dinheiro recebido: ";
    cin >> dinheiro;

    valorProduto = preco * qtd;

    if (valorProduto > dinheiro) {
        troco = valorProduto - dinheiro;
        cout << fixed << setprecision(2);
        cout << "DINHEIRO INSUFICIENTE. FALTAM " << troco << " REAIS" << endl;
    } else {
        troco = dinheiro - valorProduto;
        cout << fixed << setprecision(2);
        cout << "TROCO = " << troco << endl;
    }

    return 0;
}

File: java/troco_verificado.java
import java.util.Locale;
import java.util.Scanner;

public class troco_verificado {

	public static void main(String[] args) {
		
		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
			
		int quantidade;
		double preco, dinheiro , troco, valor, restante;
		
		System.out.print("Preço unitário do produto: ");
		preco = sc.nextDouble();
		System.out.print("Quantidade comprada: ");
		quantidade = sc.nextInt();
		System.out.print("Dinheiro recebido: ");
		dinheiro = sc.nextDouble();
		
		valor = preco * quantidade;
		
		if (dinheiro > valor) {
			troco = dinheiro - valor;
			System.out.println("TROCO = " + String.format("%.2f", troco));
		} else {
			restante = valor - dinheiro;
			System.out.println("DINHEIRO INSUFICIENTE. FALTAM " + String.format("%.2f", restante) 
								+ " REAIS");
		}
		
		sc.close();
	}

}
